src/config.c

```diff
@@ -832,9 +832,9 @@ char *cfg_escape(const char *s) {
 
 static int cmp_comp(const void *v1, const void *v2) {
   const Component_t *c1 = v1, *c2 = v2;
-  int ret = strcmp(c1->name, c2->name);
+  int ret = str_eq(c1->name, c2->name);
```

Comment:
Uhmm, no! strcmp() is used here with its  tri-state range of negative, positive, 0 (will be used for sorting, and -1 *differs* from 1 in its effect). Cannot collapse its result here to Boolean what str_eq() would do. Please revert *both* changes in this function.

Reply:
Sorry, I didn't realize that returning a negative number was key here.

Reply:
All good, it's hard to know the intention of code. That's what makes wholesale replacements of strcmp() et al so tricky! But I agree with you that it's worth our while as it improves readability (almost everywhere) and I have come to enjoy using str_*() in new code.

Reply:
> All good, it's hard to know the intention of code.

Yes, this one was a bit hidden. I did a "blind" replacement because the code looked obvious.